content-services/latest/config/mtls.md

```diff
@@ -0,0 +1,283 @@
+|alfrescoformat|classic/current|The default format for certificates is `current` for IE SS 2.0.0+ and `classic` for previous versions. The main difference is that classic format generates text files containing keystore/truststore password, private key/public key alias, and in case of keystore, also the private key password (which currently has to be the same as the keystore). The default value is `current`.|
+
+> **Note:** If you plan to use custom DNames in your certificates, you must use double quotes around the values. For example:
+
+```bash
+$ ./run.sh -cacertdname  
+"/C=GB/ST=UK/L=Maidenhead/O=Alfresco/OU=Unknown/CN=Windows Alfresco CA"
+-repocertdname "/C=GB/ST=UK/L=Maidenhead/O=Alfresco/OU=Unknown/CN=Repo"
+-solrcertdname "/C=GB/ST=UK/L=Maidenhead/O=Alfresco/OU=Unknown/CN=Solr"
+```
+
+It is recommended that you set your own passwords when generating certificates. For example:
+
+(For Linux)
+
+```bash
+./run.sh -keystorepass "password" -truststorepass "password"
+```
+
+(For Windows)
+
+```bash
+run.cmd -keystorepass "password" -truststorepass "password"
+```
+
+### Additional script examples
+
+Find more sample scripts in [alfresco-ssl-generator](https://github.com/Alfresco/alfresco-ssl-generator) repository [for Linux](https://github.com/Alfresco/alfresco-ssl-generator/tree/master/ssl-tool/samples) and [for Windows](https://github.com/Alfresco/alfresco-ssl-generator/tree/master/ssl-tool-win/samples).
+
+> **Note:** In Windows:
+>    * To run sample scripts you’ll need to move them 1 directory upwards
+>    * Add keytool to PATH system variable
+>    * Set 2 environment variables: `SET OPENSSL_CONF=<path_to_project>\alfresco-ssl-generator\ssl-tool-win\openssl.cnf` and `SET RANDFILE=.rnd`
+
+## Keystore directory structure
+
+The `keystores` directory contains the following structure and files:
+
+```bash
+keystores
+├── alfresco
+│   ├── keystore
+│   ├── ssl.keystore
+│   ├── ssl.truststore
+├── client
+│   └── browser.p12
+└── solr
+│   ├── ssl-repo-client.keystore
+│   └── ssl-repo-client.truststore
+```
+
+|File name|Description|
+|---------|-----------|
+|browser.p12|The PKCS12 keystore generated from `ssl.keystore` that contains the repository private key and certificate for use in browsers, such as Firefox.|
+|keystore|Secret key keystore containing the secret key used to encrypt and decrypt node properties.|
+|ssl.keystore|Repository keystore containing the repository private/public key pair and certificate.|
+|ssl.truststore|Repository truststore containing certificates that the repository trusts.|
+|ssl-repo-client.keystore|Solr SSL keystore containing the Solr private/public key pair and certificate.|
+|ssl-repo-client.truststore|Solr truststore containing certificates that the repository trusts.|
+
+## Repository configuration

```

Comment:
Here is the weird part, Solr/Elasticsearch and Transform Services have 2 different approaches to configuration (until the follow-up technical issue is taken care of and only the new approach stays). Which would mean that it might be best to split this section into something like below:

Repository configuration
    Common (client and server keystore setting, server is set on Tomcat)
    Solr specific (solr properties that need to be set on Repository)
    Transform Service specific (httpClient properties for transform service that need to be set on Repository)
Configuration for Solr (same level as Repository configuration)
Configuration for Transform Services (Done)

Reply:
I'd ask of you to copy the content of current InsightEngine documentation here as common + solr specific configuration. 
I'd be able to then edit this to separate the common from the solr specific and also get a Configuration for Solr section filled, to reach the state which I recommended earlier :)

Reply:
I will work on this in another change, so we can hopefully move forward with publishing what we have.